### PR TITLE
Feature/filter dog list

### PR DESCRIPTION
**Pull request by clacayetano**
- Adds GET /api/breeds endpoint
- Adds breeds array in store
- Add a filter side bar inside Dogs component
- Makes Dogs a class component so that it can have state
- Filter dogs by breed

Just looking for early feedback but this is a mergeable PR too. Will continue working on filtering by age, gender and size.

### CODE REVIEW

File: client/Components/Dogs/Dogs.jsx [hunk 1]
```diff
@@ -1,20 +1,41 @@
...
-    {dogs.map(dog => (
-      <div key={dog.id}>
-        <Link to={`${DOGS}/${dog.id}`}>{dog.name}</Link> {dog.price}{' '}
-        {dog.description}
+class Dogs extends Component {
+  constructor() {
+    super();
+    this.state = {
+      breedId: 0
+    };
```
Comment (inKEYYgnito) [inline client/Components/Dogs/Dogs.jsx:14]: just aesthetic, but u can take off constructor and just this.state = {}

File: client/Components/FilterSideBar.jsx [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+    <>
+      <select value={breedId} onChange={onChange}>
+        <option key={0} value={0}>
+          -- Any Breed --
+        </option>
+        {breeds.map(breed => (
+          <option key={breed.id} value={breed.id}>
+            {breed.name}
+          </option>
+        ))}
```
Comment (inKEYYgnito) [inline client/Components/FilterSideBar.jsx:20]: `[{id: 0, name: '-- Any Breed --'}, ...breeds].map() `

just a suggestion to have the other option connected

**Review by inKEYYgnito — APPROVED** [review]
added some suggestion/s, but otherwise approved.

File: client/Components/Dogs/Dogs.jsx (hunk 1 above)
Comment (clacayetano) [reply client/Components/Dogs/Dogs.jsx:14]: haven't tried this syntax yet...

**Pull request merged**